core/src/main/java/io/kestra/core/storages/StorageInterface.java

```diff
@@ -15,7 +15,24 @@
 import java.util.List;
 
 @Introspected
-public interface StorageInterface extends Plugin {
+public interface StorageInterface extends AutoCloseable, Plugin {
+
+    /**
+     * Opens any resources or perform any pre-checks for initializing this storage.
+     *
+     * @throws IOException if an error happens during initialization.
+     */
+    default void init() throws IOException {
+        // no-op
+    }
+
+    /**
+     * Closes any resources used by this class.
+     */
+    @Override
+    default void close() {
+        // no-op
+    }
```

Comment:
Not sure it's a good idea as most if not all storage may have something to close so it's better to mandate thinking about it.